dm/syncer/data_validator.go

```diff
@@ -935,3 +968,31 @@ func (v *DataValidator) getTableStatus() map[string]*tableValidateStatus {
 	}
 	return tblStatus
 }
+
+func (v *DataValidator) GetBriefValidatorStatus() *pb.BriefValidationStatus {
+	v.RLock()
+	defer v.RUnlock()
+	// todo: last check time
+	ret := &pb.BriefValidationStatus{
+		Mode:          v.cfg.ValidatorCfg.Mode,
+		LastCheckTime: v.lastCheckTime.Load().String(),
+		Stats: &pb.ValidationStats{
+			ProcessedCount: 0,
+			ErrorCount:     0,
+		},
+	}
+	pendingRowCnt := 0
+
+	for i := 0; i < len(v.processedRowCounts); i++ {
+		ret.Stats.ProcessedCount += uint64(v.processedRowCounts[i].Load())
+		pendingRowCnt += int(v.pendingRowCounts[i].Load())
+	}
+	// currently, errRowCounts has same length with processedRowCounts
+	// to avoid potential careless changes (in the future)
+	// they are iterated separately.
+	for i := 0; i < len(v.errorRowCounts); i++ {
+		ret.Stats.ErrorCount += uint64(v.errorRowCounts[i].Load())
+	}
+	ret.Stats.SucceededCount = ret.Stats.ProcessedCount - ret.Stats.ErrorCount - uint64(pendingRowCnt)
```

Comment:
seems `loadPersistedData` will load `errorRowCounts`, but other variables will not be loaded. So this calculation may get negative value.

Reply:
Indeed. `pedingRowCounts` can be recovered in `loadPersistedData`. What about adding a new checkpoint to record `processedCount`:
```sql
CREATE TABLE task_name(
id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
source VARCHAR(32),
row_change_type INT,
processed_count INT,
UNIQUE KEY uk_validator_process(source, row_change_type)
)
```
/cc @D3Hunter